Implement the ABI lookup support by enhancing the MoveStructType inte…

VolodymyrBg opened the pull request:
…rface

I've implemented the ABI lookup support by enhancing the MoveStructType interface. Here's what I added:
type: The fully qualified name of the struct in the Move format (address::module::name)
fields: A map of field names to their Move types, allowing for dynamic lookup of struct fields
genericTypeParams: Optional array of generic type parameters if the struct is generic
abilities: Optional array of Move abilities that the struct has
This implementation allows for proper typing of Move structs and enables:
Type checking of struct fields
Support for generic structs
Ability to verify struct capabilities
Dynamic field lookup from ABI

VolodymyrBg committed:
Implement the ABI lookup support by enhancing the MoveStructType interface

gregnazario reviewed (dismissed):
Would be good to add a test for it

VolodymyrBg committed:
feat(types): Implement MoveStructType with ABI support

- Add proper typing for Move struct fields and their types
- Support generic type parameters and abilities
- Add comprehensive test suite for struct type validation
- Improve type safety for Move struct handling

VolodymyrBg commented:
@gregnazario  Added test

VolodymyrBg committed:
Update types.ts

VolodymyrBg commented:
> Looks like there's some typing issues. Can you run the lints `npm lint` and `npm build` and see how to fix it?

working on it

VolodymyrBg committed:
Update moveStructType.test.ts

VolodymyrBg commented:
@gregnazario  Done.

VolodymyrBg committed:
Update moveStructType.test.ts

VolodymyrBg commented:
@gregnazario 
I hope now it's going to work properly.. sorry for taking your time

VolodymyrBg commented:
Last attempt))
@gregnazario

VolodymyrBg committed:
Update moveStructType.test.ts

VolodymyrBg commented:
Never give up

VolodymyrBg committed:
Update moveStructType.test.ts

VolodymyrBg commented:
@gregnazario  I hope it works properly now

VolodymyrBg commented:
@gregnazario
